lib/client/client.js

```diff
@@ -298,6 +298,46 @@ Client.prototype.bind = function bind (name,
   return this._send(req, [errors.LDAP_SUCCESS], null, callbackWrapper, _bypass)
 }
 
+/**
+ * Performs a sasl (NTLM/GSSAPI) authentication against the server.
+ *
+ * @param {String} token the token from the authentication http header.
+ * @param {Control} controls (optional) either a Control or [Control].
+ * @param {Function} callback of the form f(err, res).
+ * @throws {TypeError} on invalid input.
+ */
+Client.prototype.saslBind = function saslBind (token,
+  controls,
+  callback,
+  _bypass) {
+  if (typeof (token) !== 'string' && !Buffer.isBuffer(token)) {
+    throw new TypeError('token (string or Buffer) required')
+  }
+  if (typeof (controls) === 'function') {
+    callback = controls
+    controls = []
+  } else {
+    controls = validateControls(controls)
+  }
```

Comment:
I'd rather the function accept an input object so that positional parameter inspection is unnecessary.

Reply:
Are you talking about an incopatible change also for `client.bind()`?

Reply:
No. This is a completely new method. There is no reason to carry forward difficult to maintain patterns.